src/content/__tests__/monetizationLinkManager.test.ts

```diff
@@ -285,9 +334,255 @@ describe('MonetizationLinkManager', () => {
+
+    test('should handle element with link tag append and removal', async () => {
+      const { document, documentReadyState } = createTestEnv({});
+
+      const linkManager = createMonetizationLinkManager(document);
+      documentReadyState.mockReturnValue('interactive');
+
+      msg.GET_WALLET_ADDRESS_INFO.mockResolvedValueOnce(
+        success(WALLET_ADDRESS[0]),
+      );
+
+      linkManager.start();
+
+      const wrapper = document.createElement('div');
+      wrapper.innerHTML =
+        '<link rel="monetization" href="https://ilp.interledger-test.dev/tech">';
+      document.head.appendChild(wrapper);
+
+      await new Promise(process.nextTick);
+
+      const requestId = requestIdMock.mock.results[1].value;
+      expect(msg.START_MONETIZATION).toHaveBeenCalledWith([
+        {
+          requestId,
+          walletAddress: WALLET_ADDRESS[0],
+        },
+      ]);
+
+      wrapper.remove();
+
+      await new Promise(process.nextTick);
+
+      expect(msg.STOP_MONETIZATION).toHaveBeenCalledWith([
+        {
+          requestId,
+          intent: 'remove',
+        },
+      ]);
+    });
+
+    test('more link tags added right after, leading to another mutation', async () => {
+      const { document, documentReadyState } = createTestEnv({});
+
+      const linkManager = createMonetizationLinkManager(document);
+      documentReadyState.mockReturnValue('interactive');
+
+      msg.GET_WALLET_ADDRESS_INFO.mockResolvedValueOnce(
+        success(WALLET_ADDRESS[0]),
+      )
+        .mockResolvedValueOnce(success(WALLET_ADDRESS[1]))
+        .mockResolvedValueOnce(success(WALLET_ADDRESS[2]));
+
+      linkManager.start();
+
+      const link1 = document.createElement('link');
+      link1.rel = 'monetization';
+      link1.href = 'https://ilp.interledger-test.dev/tech1';
+      document.head.appendChild(link1);
+
+      const link2 = document.createElement('link');
+      link2.rel = 'monetization';
+      link2.href = 'https://ilp.interledger-test.dev/tech2';
+      document.head.appendChild(link2);
+
+      const link3 = document.createElement('link');
+      link3.rel = 'monetization';
+      link3.href = 'https://ilp.interledger-test.dev/tech3';
+      document.head.appendChild(link3);
+
+      await new Promise(process.nextTick);
+
+      const request1Id = requestIdMock.mock.results[1].value;
+      const request2Id = requestIdMock.mock.results[2].value;
+      const request3Id = requestIdMock.mock.results[3].value;
+
+      expect(msg.GET_WALLET_ADDRESS_INFO).toHaveBeenCalledTimes(3);
+      expect(msg.START_MONETIZATION).toHaveBeenCalledTimes(1);
+      expect(msg.START_MONETIZATION).toHaveBeenCalledWith([
+        {
+          requestId: request1Id,
+          walletAddress: WALLET_ADDRESS[0],
+        },
+        {
+          requestId: request2Id,
+          walletAddress: WALLET_ADDRESS[1],
+        },
+        {
+          requestId: request3Id,
+          walletAddress: WALLET_ADDRESS[2],
+        },
+      ]);
```

Comment:
In this test, we want to see two `START_MONETIZATION` calls, one from each mutation callback.